libsmack/common.c

```diff
@@ -204,3 +205,64 @@ int apply_cipso(const char *path)
 
 	return 0;
 }
+
+int apply_onlycap(const char *path)
+{
+	FILE *file = fopen(path, "r");
+	int ret = 0;
+	if (file == NULL) {
+		if (errno == ENOENT) /* there is no configuration file,
+				        we should do nothing */
+			return 0;
+		fprintf(stderr, "open() failed for '%s' : %s\n", path,
+			strerror(errno));
+		return -1;
+	}
+
+	char buf[SMACK_LABEL_LEN + 2];
+	int cnt = 0;
+	int size = 10;
+	char **labels = malloc(sizeof(char *) * size);
+	if (labels == NULL) {
+		fputs("Out of memory.\n", stderr);
+		ret = -1;
+		goto out;
+	}
+
+	while (fgets(buf, SMACK_LABEL_LEN + 2, file) != NULL) {
```

Comment:
There is an established convention in this library:
- Public functions declared in libsmack.c implement interfacing with Smack and also reading and parsing configuration data from files
- Internal apply_* functions declared in common.c implement policy loading by calling relevant functions from libsmack.

Therefore I would prefer to move the code reading and parsing Smack labels from onlycap file to a separate public libsmack function - for example `smack_set_onlycap_from_file`. Then it would be called here, in `apply_onlcap`.